classes/controllers/grid/GridHandler.inc.php

```diff
@@ -390,7 +392,10 @@ public function setGridDataElements($data)
     {
         $this->callFeaturesHook('setGridDataElements', ['grid' => &$this, 'data' => &$data]);
 
-        if (is_iterable($data)) {
+        // Check for Enumerable first as it would pass is_iterable test
+        if ($data instanceof Enumerable) {
+            $this->_data = EntityDAO::toAssociativeArray($data);
```

Comment:
I think you should be able to use iterator_to_array here with the second parameter set to `true`, if your DAO yields from `getMany` using keys, e.g.:
```
yield $row->issue_id => $this->fromRow($row);
```
That would mean `toAssociativeArray` can be removed in favour of standard built-ins.